plugin/pkg/scheduler/factory/factory.go

```diff
@@ -1014,6 +1014,12 @@ func (p *podPreemptor) UpdatePodAnnotations(pod *v1.Pod, annotations map[string]
 	for k, v := range annotations {
 		podCopy.Annotations[k] = v
 	}
-	_, err := p.Client.CoreV1().Pods(podCopy.Namespace).UpdateStatus(podCopy)
+	annotationArray := make([]string, 0, len(podCopy.Annotations))
+	for k, v := range podCopy.Annotations {
+		annotationArray = append(annotationArray, "\""+k+"\":\""+v+"\"")
+	}
+	annotationString := "{" + strings.Join(annotationArray, ",") + "}"
+	patch := fmt.Sprintf(`{"metadata":{"annotations":%v}}`, annotationString)
```

Comment:
This is not safe - if annotation contains quotes and/or slashes it will invalid json. A better approach might be to create an anonymous struct and call `json.Marshal(...)` on it.

Reply:
Thanks for your suggest. I have fix it.